### PR TITLE
Adding delete button to allow users to delete comments

### PR DESCRIPTION
**Pull request by majid-L**
Adding some restrictions and a confirmation message to let users know when a comment has been deleted

### CODE REVIEW

File: src/Components/VoteUpdateButtons.jsx [hunk 1]
```diff
@@ -1,16 +1,30 @@
...
 const handleClick = () => 
     !loggedInUser ? () => setNotLoggedInError(true) : updateVotes;
 
+const handleDelete = () => {
+  setDisableDelete(true);
+  api.deleteComment(id).then(() => {
+    setComments(prev => prev.filter(comment => 
+      comment.comment_id !== id));
+    setShowDeletedMsg(true);
+    setTimeout(() => setShowDeletedMsg(false), 5000);
```
Comment (popatre) [inline src/Components/VoteUpdateButtons.jsx]: What if it doesnt successfully delete?

**Review by popatre — CHANGES_REQUESTED** [review]
Most of the way there, some stuff to add:

- I'd ideally only want to be able to delete my own comments, not everyone can delete everyone elses
- At the moment, if the api request doesnt work, it just constantly says `deleting...` despite the fact its failed. 
- Could you disable the button after its been clicked. Just so i cant keep clicking it on a slow internet connection?

-

File: src/App.js [hunk 2]
```diff
@@ -21,12 +21,16 @@ function App() {
...
       <Routes>
         <Route path="/" element={<Home />}/>
-        <Route path="/articles" element={<Articles />}/> 
-        <Route path="/:topic/articles" element={<Articles />}/> 
+        <Route path="/articles" element={<Articles />}/>
         <Route path="/articles/:articleId" element={<SingleArticle loggedInUser={loggedInUser} />}/> 
         <Route path="/articles/:articleId/comments" element={<Comments loggedInUser={loggedInUser} />} />
       </Routes>
       </div>
+      {useLocation().pathname !== '/' && <button onClick={() => window.scrollTo({
```
Comment (roisinoreilly) [inline src/App.js:29]: very nice

File: src/Components/Buttons.jsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+const Buttons = ({votes, showError, updateVotes, loggedInUser, author, id, setComments, setShowDeletedMsg, setShowWrongUserMsg, setShowDeleteError}) => {
+  
+const [notLoggedInError, setNotLoggedInError] = useState(false);
+const [disableDelete, setDisableDelete] = useState(false);
+const [isLoading, setIsLoading] = useState(false);
+
+const handleClick = () => 
+    !loggedInUser ? () => setNotLoggedInError(true) : updateVotes;
+
+const handleDelete = () => {
```
Comment (roisinoreilly) [inline src/Components/Buttons.jsx:13]: I think that we shouldn't be able to see a delete button for comments that aren't the logged in user, not just an error message

File: src/Components/Comments.jsx [hunk 4]
```diff
@@ -16,8 +17,18 @@ const [showPostButton, setshowPostButton] = useState(false);
...
+const [showDeletedMsg, setShowDeletedMsg] = useState(false);
+
 const [showErrorMsg, setShowErrorMsg] = useState(false);
 const [showNotLoggedIn, setshowNotLoggedIn] = useState(false);
+const [showWrongUserMsg, setShowWrongUserMsg] = useState(false);
+const [showDeleteError, setShowDeleteError] = useState(false);
+
+const myRef = useRef(null);
+
+if (showDeletedMsg || showWrongUserMsg || showDeleteError) {
```
Comment (roisinoreilly) [inline src/Components/Comments.jsx]: really nicely handled

**Review by roisinoreilly — APPROVED** [review]
really excellent work, the only thing I would suggest for this ticket is that the user shouldn't see a delete button for comments that aren't theirs, otherwise merge when ready

**Pull request merged**